### ISSUE TITLE
(xQueueGenericReceive)- assert failed! when calling connect(client)

### ISSUE THREAD

**Issue by EHR11** [knolleary/pubsubclient#541]
Hello,

I'm in the process of putting together a LoRaWAN node with MQTT capabilities, using a WiFi LoRa 32 (SX1278 board)

Trying to connect to my broker gives me this error: 

`/Users/ficeto/Desktop/ESP32/ESP32/esp-idf-public/components/freertos/queue.c:1441 (xQueueGenericReceive)- assert failed!
abort() was called at PC 0x400858e9 on core 1

Backtrace: 0x40088c78:0x3ffb1cf0 0x40088e4d:0x3ffb1d10 0x400858e9:0x3ffb1d30 0x400f97a2:0x3ffb1d70 0x400f99c2:0x3ffb1d90 0x400ed5d8:0x3ffb1db0 0x400ed641:0x3ffb1dd0 0x400fad6a:0x3ffb1df0 0x400f9cbc:0x3ffb1e10 0x400ebd9c:0x3ffb1e50 0x400d5b81:0x3ffb1e70 0x400d1936:0x3ffb1eb0 0x400d1b1a:0x3ffb1f10 0x400d211d:0x3ffb1f40 0x400d226d:0x3ffb1f90 0x400fcfee:0x3ffb1fb0 0x40084ce9:0x3ffb1fd0
`

Here's the function where the problem originates:

`void reconnect() {
  // Loop until we're reconnected
  while (!mqtt_client.connected()) {
    println_oled("Making MQTT connection");
    // Create a random client ID
    String clientId = "ESP32Client-";
    clientId += String(random(0xffff), HEX);
    // Attempt to connect
    if (mqtt_client.connect(clientId.c_str())) {
      println_oled("Connected!");
      // Once connected, publish an announcement...
      //mqtt_client.publish("outTopic", "hello world");
      // ... and resubscribe
      mqtt_client.subscribe(mqtt_topic);
    } else {
      print_oled("Failed, RC=");
      print_oled(String(mqtt_client.state()));
      println_oled("");
      println_oled("Try again in 5 seconds");
      // Wait 5 seconds before retrying
      delay(5000);
    }
  }
}` 

Is this avoidable? Am I missing something? 

Thanks in advance.

**Comment by etawyy** [comment]
Hi,
I ran into the same issue, connecting (over Wifi) to a mqtt broker after a period of sleep.
Did you find a solution to your problem?
Thanks!

**Comment by jwolansk** [comment]
In my case an initial connection to the server causes this error. Could it be multi threading related?

**Comment by etawyy** [comment]
well I seem to have solved it in my case by: 
- arranging properly the code as says the documentation: 1-callback header, 2-client constructor, 3-callback function
- making sure I was properly connected to wifi before calling the publish function.

**Comment by jwolansk** [comment]
Yeah, turned out there was no wifi available during my connect call. Thanks for responding!

**Comment by PIYUSHGIRI2021** [comment]
hello @Alvin1Zhang and every one ...i compile pppos_client example in esp idf before some day i copiled it will work fine but not its not work,please let me know how to resolve below error..

I (24525) pppos_example: Modem PPP Started
C:/Users/Frank/Desktop/esp-idf/components/freertos/queue.c:1448 (xQueueGenericReceive)- assert failed!

abort() was called at PC 0x40088345 on core 0
0x40088345: xQueueGenericReceive at C:/Users/Frank/Desktop/esp-idf/components/freertos/queue.c:1448 (discriminator 3)

Backtrace:0x40084f7f:0x3ffb7ab0 0x4008561d:0x3ffb7ad0 0x4008928e:0x3ffb7af0 0x40088345:0x3ffb7b60 0x400f04a5:0x3ffb7ba0 0x400f07bc:0x3ffb7bc0 0x400deb80:0x3ffb7be0 0x400dec0c:0x3ffb7c00 0x400df989:0x3ffb7c20 0x400f4c5c:0x3ffb7c50 0x400f4e9c:0x3ffb7c80 0x400f55c5:0x3ffb7ca0 0x400f34f2:0x3ffb7cc0 0x400f3569:0x3ffb7ce0 0x400f3575:0x3ffb7d00 0x400f36a5:0x3ffb7d20 0x400eed19:0x3ffb7d40 0x400eed4e:0x3ffb7d60 0x400ee0a1:0x3ffb7d80 0x400eea08:0x3ffb7db0 0x400eeb5d:0x3ffb7dd0 0x400dd2d1:0x3ffb7df0 0x400dd350:0x3ffb7e10 0x40085625:0x3ffb7e40
0x40084f7f: panic_abort at C:/Users/Frank/Desktop/esp-idf/components/esp_system/panic.c:330

0x4008561d: esp_system_abort at C:/Users/Frank/Desktop/esp-idf/components/esp_system/system_api.c:106

0x4008928e: abort at C:/Users/Frank/Desktop/esp-idf/components/newlib/abort.c:46
